packages/-ember-data/tests/unit/promise-proxies-test.js

```diff
@@ -110,3 +113,69 @@ module('PromiseManyArray', function() {
     });
   });
 });
+
+module('unit/PromiseBelongsTo', function(hooks) {
+  setupTest(hooks);
+
+  class Parent extends Model {
+    @belongsTo('child', { async: true })
+    child;
+  }
+  class Child extends Model {
+    @belongsTo('parent', { async: false })
+    parent;
+  }
+  class ChildAdapter extends Adapter {
+    findRecord(store, type, id, snapshot) {
+      const ChildRecord = {
```

Comment:
Do we know if accessing meta ends up triggering this request? We should add an assertion to codify the current behavior so we at least know if/when we change it later.

Reply:
Yeah, in the console it was trying to fetch the record but the test would fail because the endpoint doesn't exist so i just mocked it. 
I added some debuggers in and it appears that findRecord runs before we check the meta, so i guess the `parent.child` bit causes the request to trigger.

Reply:
That's what I'd expect. We should add a test so that this behavior is explicitly covered. It's not necessarily a good behavior, but knowing that's what we do now is important to refactoring later :)